test/tsan.suppress

```diff
@@ -0,0 +1,10 @@
+# ThreadSanitizer suppressions file for realm-core
+
+# `AESCryptor::read()` and`copy_up_to_date_page()` copy entire pages.
```

Comment:
Rather than suppressing the correct data race errors, these functions could use atomic reads and writes.

While it is hard to imagine how self-assignment could break, it is nonetheless UB and this could easily mask other bugs as well.